source/features/revert-file.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate, {DelegateEvent} from 'delegate-it';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import {getOwnerAndRepo, getRepoURL} from '../libs/utils';
+
+async function handleRevertFileClick(event: React.MouseEvent<HTMLButtonElement>): Promise<void> {
+	const menuItem = event.currentTarget;
+	menuItem.textContent = 'Reverting…';
+	event.preventDefault();
+	event.stopPropagation();
+
+	const {ownerName, repoName} = getOwnerAndRepo();
+	const [, prNumber]: string[] = /pull[/](\d+)[/]files/.exec(location.pathname) || [];
+	try {
+		// Get the real base commit of this PR, not the HEAD of base branch
+		const {repository: {pullRequest: {baseRefOid}}} = await api.v4(`{
+			repository(owner: "${ownerName}", name: "${repoName}") {
+				pullRequest(number: ${prNumber}) {
+					baseRefOid
+				}
+			}
+		}`);
+
+		const filePath = (menuItem.closest('[data-path]') as HTMLElement).dataset.path!;
+		const [, forkRepoURL, forkBranch]: string[] = /^([^:]+):(.+)$/.exec(select('.head-ref')!.title) || [];
+		const [originalFile, forkFile] = await Promise.all([
+			api.v3(`repos/${getRepoURL()}/contents/${filePath}?ref=${baseRefOid}`, {ignoreHTTPStatus: true}),
+			api.v3(`repos/${forkRepoURL}/contents/${filePath}?ref=${forkBranch}`, {ignoreHTTPStatus: true})
```

Comment:
Here I’m using v3 to get the file because:

- I can’t find the API to save the file in v4, so I have to save with v3
- it’s not a good idea to GET with v4 and PUT with v3 because one returns text and the other wants base64: the maybe they treat data differently as well

If you know how to commit files with v4 let me know